bar_units.py

```diff
@@ -52,25 +54,64 @@
 ] + default_filters,
 """
 
+
+def setup():
+    if not os.path.exists('generated'):
+        os.mkdir('generated')
+
+    output = subprocess.check_output("git --version", shell=True)
+    major = re.search("(\d+)\.(\d+)\.(\d+)", str(output)).group(1)
+    minor = re.search("(\d+)\.(\d+)\.(\d+)", str(output)).group(2)
+    patch = re.search("(\d+)\.(\d+)\.(\d+)", str(output)).group(3)
+
+    if int(major) < 2 or int(minor) < 25:
+        # you can try it with an older version git but there are no guarantees
+        raise Exception(
+            "you need a newer version of git to run this program at least 2.25.0. see code for a work around")
```

Comment:
I'd like to see a better explanation of how to fix the issue in the message, especially as it doesn't say where in the code to look.